src/unitxt/metrics.py

```diff
@@ -82,43 +83,90 @@ def _can_compute_confidence_intervals(self, num_predictions):
 
+        Unlike GlobalMetric, this is simply a function of the instance scores (possibly taking into account additional_inputs field),
+         so they don't need to be recomputed after every bootstrap draw.
+
         score_names: List[str]
             Compute a confidence interval for each score_name from this list.
         instances:
             The instances for which the confidence intervals are computed.
+        aggregation_func:
+            A function with arguments instances, field_name; is applied on list of instances (which may include additional_inputs
+            field, as well as the prediction and references), and the field_name; default is simply to take the mean field_name from
+            instances after resampling, if aggregation_func=None.
         """
-        from statistics import mean
-
         result = {}
 
         if not self._can_compute_confidence_intervals(num_predictions=len(instances)):
             return result
 
-        score_names = (
-            self.ci_scores if self.ci_scores is not None else [self.main_score]
-        )
+        if score_names is None:
+            score_names = (
+                self.ci_scores if self.ci_scores is not None else [self.main_score]
+            )
+        if aggregation_func is None:
+
+            def aggregation_func(instances, field_name):
+                return mean([instance[field_name] for instance in instances])
 
         for score_name in score_names:
-            scores = [
-                instance["score"]["instance"][score_name] for instance in instances
-            ]
+            # need to redefine the statistic function within the loop because score_name is a loop variable, to avoid ruff errors
+            def statistic(arr, axis, score_name=score_name):
+                # arr is a 2d array where each row is a resampling, so we
+                # iterate over the rows and compute the metric on each resampling
+                # def metric(instances):
+                #     return aggregation_func(instances, score_name)
+
+                scores = numpy.apply_along_axis(
+                    lambda resamled_instances: aggregation_func(
+                        resamled_instances, score_name
+                    ),
+                    axis=axis,
+                    arr=arr,
+                )
+
+                return self.resample_from_non_nan(scores)
+
+            # apply bootstrap only on the relevant field
             ci = bootstrap(
-                (scores,),
-                statistic=mean,
+                (instances,),
+                statistic=statistic,
                 n_resamples=self.n_resamples,
                 confidence_level=self.confidence_level,
                 random_state=self.new_random_generator(),
             ).confidence_interval
-            result[f"{score_name}_ci_low"] = ci.low
-            result[f"{score_name}_ci_high"] = ci.high
+            full_score_name = (
```

Comment:
isn't this just `field_name_full` from https://github.com/IBM/unitxt/pull/452/files#diff-5623e0c8548470ce6339f802312be342978bc20e265c170fab1722acbbd5154eR487 or from https://github.com/IBM/unitxt/pull/452/files#diff-5623e0c8548470ce6339f802312be342978bc20e265c170fab1722acbbd5154eR484 ?

Reply:
yes but because this is called on the same instance set for both the group_mean and mean reductions, I want to differentiate between the CI for these two reductions, though the score_name operated on is the same for both reductions.  Therefore I pass the aggregation prefix for the group_mean reduction and an empty string if is just the CI of the mean of that score